### PR TITLE
Enable NetworkIsolationKey-based partitions.

### PR DESCRIPTION
**Pull request by goodov**
https://github.com/brave/brave-browser/issues/11816

Enable the same features Chromium currently is rolling out. At first test it on nightly.

### CODE REVIEW

File: seed/seed.json [hunk 1]
```diff
@@ -1074,6 +1074,34 @@
...
+        {
+            "name": "PartitionConnectionsByNetworkIsolationKeyStudy",
+            "experiments": [
+                {
+                    "name": "Enabled",
+                    "probability_weight": 100,
+                    "feature_association": {
+                        "enable_feature": [
+                            "PartitionConnectionsByNetworkIsolationKey",
+                            "PartitionDomainReliabilityByNetworkIsolationKey",
```
Comment (iefremov) [inline seed/seed.json:1087]: @pes10k can you please confirm we need `PartitionDomainReliabilityByNetworkIsolationKey`? just not sure we have it enabled

**Comment by iefremov** [comment]
@goodov there is a conflict

File: seed/seed.json (hunk 1 above)
Comment (pes10k) [reply seed/seed.json:1087]: I think you're right, we shouldn't have the domain reliability functionality enabled (if we haven't already disabled it), and so we don't need the `PartitionDomainReliabilityByNetworkIsolationKey` feature either

Comment (pes10k) [reply seed/seed.json:1087]: https://chromium.googlesource.com/chromium/src/+/refs/heads/main/components/domain_reliability/README.md if others aren't familiar with what the feature does (it reports to google when google properties are down)

Comment (goodov) [reply seed/seed.json]: agree. the domain reliability feature is disabled on our end.

**Review by iefremov — APPROVED** [review]

**Pull request merged**